dash/include/dash/Lazy.h

```diff
@@ -0,0 +1,77 @@
+#ifndef DASH__LAZY_H__INCLUDED
+#define DASH__LAZY_H__INCLUDED
+
+#include <functional>
+
+
+namespace dash {
+
+/* TODO:
+ * Should be
+ * 
+ *   template <typename T, bool evaluated>
+ *   class Lazy;
+ *
+ * with:
+ *
+ *   constexpr Lazy<T, true>
+ *   Lazy<T, false>::operator T() const {
+ *     return Lazy<T, true>(gen_fun());
+ *   }
+ *
+ * and:
+ *
+ *   constexpr Lazy<T, true>
+ *   Lazy<T, true>::operator T() const {
+ *     return _value;
+ *   }
+ * 
+ *
+ */
+
+template <typename T>
+class Lazy {
+
```

Comment:
`dash::Deferred` is a wrapper / expression type and not related to async, I think you are referring to the policy `dash::launch::deferred` (corresponding to `std::launch::deferred`, if it will be introduced)